### PR TITLE
feat: Add Version to Kyma.Spec.Modules

### PR DESCRIPTION
**Pull request by ruanxin**
related issue: https://github.com/kyma-project/lifecycle-manager/issues/1589

### CODE REVIEW

File: api/v1beta2/kyma_types.go [hunk 1]
```diff
@@ -211,14 +219,6 @@ type PartialMeta struct {
 
 const DefaultChannel = "regular"
 
-func PartialMetaFromObject(object apimetav1.Object) PartialMeta {
```
Comment (ruanxin) [inline api/v1beta2/kyma_types.go:214]: To reviewer:

the reason to remove this function is make the `Generation` assignment trackable by IDE, with this `Object` as general input, we lost track of when the generation get assigned [here](https://github.com/kyma-project/lifecycle-manager/pull/1694/files#diff-01b10c0b6b181efc907c8638a4f6af573038dd730f8b5130be1b7071f5fba6f2R304-R306)

File: internal/controller/kyma/controller.go [hunk 2]
```diff
@@ -529,18 +524,6 @@ func (r *Reconciler) updateStatusWithError(ctx context.Context, kyma *v1beta2.Ky
 	return nil
 }
 
-func (r *Reconciler) GenerateModulesFromTemplate(ctx context.Context, kyma *v1beta2.Kyma) (common.Modules, error) {
-	lookup := templatelookup.NewTemplateLookup(client.Reader(r), r.DescriptorProvider)
-	templates := lookup.GetRegularTemplates(ctx, kyma)
-	for _, template := range templates {
-		if template.Err != nil {
-			r.Event.Warning(kyma, moduleReconciliationError, template.Err)
```
Comment (ruanxin) [inline internal/controller/kyma/controller.go:537]: To reviewer:
this event makes no sense.

File: config/crd/bases/operator.kyma-project.io_kymas.yaml [hunk 3]
```diff
@@ -64,6 +64,7 @@ spec:
                       description: |-
                         Channel is the desired channel of the Module. If this changes or is set, it will be used to resolve a new
                         ModuleTemplate based on the new resolved resources.
+                        The Version and Channel are mutually exclusive options.
```
Comment (Tomasz-Smelcerz-SAP) [inline config/crd/bases/operator.kyma-project.io_kymas.yaml:67]: Why is the version `v1beta1` extended?

File: tests/integration/controller/kyma/kyma_module_channel_test.go [hunk 4]
```diff
@@ -76,7 +76,7 @@ var _ = Describe("module channel different from the global channel", func() {
 	})
 })
 
-var _ = Describe("Given invalid channel", func() {
+var _ = Describe("Given invalid channel which rejected by CRD validation rules", func() {
```
Comment (Tomasz-Smelcerz-SAP) [inline tests/integration/controller/kyma/kyma_module_channel_test.go:79]:
```suggestion
var _ = Describe("Given invalid channel which is rejected by CRD validation rules", func() {
```

File: tests/integration/controller/kyma/kyma_module_version_test.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var _ = Describe("Given invalid module version which rejected by CRD validation rules", func() {
```
Comment (Tomasz-Smelcerz-SAP) [inline tests/integration/controller/kyma/kyma_module_version_test.go:12]:
```suggestion
var _ = Describe("Given invalid module version which is rejected by CRD validation rules", func() {
```

File: config/crd/bases/operator.kyma-project.io_kymas.yaml (hunk 3 above)
Comment (ruanxin) [reply config/crd/bases/operator.kyma-project.io_kymas.yaml:67]: because most v1beta1 fields are from v1beta2 if they are consistent.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Pull request merged**